linera-core/src/client.rs

```diff
@@ -151,29 +144,37 @@ where
         pending_blobs: BTreeMap<BlobId, HashedBlob>,
     ) -> ChainClient<P, S>
     where
-        P: Clone,
         ViewError: From<S::ContextError>,
     {
         let known_key_pairs = known_key_pairs
             .into_iter()
             .map(|kp| (Owner::from(kp.public()), kp))
             .collect();
-        ChainClient {
-            client: self.clone(),
-            chain_id,
+
+        use dashmap::mapref::entry::Entry;
```

Comment:
nit: How about inlining the `use` since `Entry` is used only once?

Reply:
Good idea!  I was planning to use the other variant as well but ended up not.